dali/pipeline/data/buffer.h

```diff
@@ -37,6 +37,22 @@ namespace dali {
 class GPUBackend;
 class CPUBackend;
 
+/**
+ * @brief State of a batch object. Can be used to pin the state to one of the two contiguous /
+ * noncontiguous modes or to change the mode when resizing, if it was not pinned.
+ * See TensorVector::Resize for detailed behaviour.
+ * * Default - means we are leaving it to the Batch to decide when to coalesce the allocation,
+ *   or keep the current one unchanged.
+ * Contiguous and Noncontiguous allow to request one backing allocation or separate ones for each
+ * sample.
+ */
+enum class BatchState {
```

Comment:
s/BatchState/BatchContiguity/? And same for accesors.

Reply:
Done. Although, now I feel a bit repetitive with SetContiguity(BatchContiguity) due to the length of the world.